internal/driver/basenotificationresthandler.go

```diff
@@ -0,0 +1,145 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	sdk "github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+
+	"github.com/IOTechSystems/onvif/event"
+	"github.com/IOTechSystems/onvif/gosoap"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	OnvifEventRestPath = "onvifevent"
+	apiResourceRoute   = common.ApiBase + "/" + OnvifEventRestPath + "/{" + common.DeviceName + "}/{" + common.ResourceName + "}"
+	handlerContextKey  = "RestHandler"
+)
+
+type RestHandler struct {
+	service     *sdk.DeviceService
+	logger      logger.LoggingClient
+	asyncValues chan<- *models.AsyncValues
+}
+
+func NewRestHandler(service *sdk.DeviceService, logger logger.LoggingClient, asyncValues chan<- *models.AsyncValues) *RestHandler {
+	handler := RestHandler{
+		service:     service,
+		logger:      logger,
+		asyncValues: asyncValues,
+	}
+
+	return &handler
+}
+
+func (handler RestHandler) Start() errors.EdgeX {
```

Comment:
Start to me indicates long running...
```suggestion
func (handler RestHandler) AddRoute() errors.EdgeX {
```

Reply:
Renamed the func.